runtimes/peregrine/src/lib.rs

```diff
@@ -852,6 +856,9 @@ impl did::DeriveDidCallAuthorizationVerificationKeyRelationship for Call {
 			Call::Attestation(_) => Some(did::DidVerificationKeyRelationship::AssertionMethod),
 			Call::Ctype(_) => Some(did::DidVerificationKeyRelationship::AssertionMethod),
 			Call::Delegation(_) => Some(did::DidVerificationKeyRelationship::CapabilityDelegation),
+			// DID creation is not allowed through the DID proxy.
```

Comment:
Just wondering what can happen if we allow this?

Reply:
The verification prior to dispatch would always fail with a `DidNotPresent` error as there's no DID in the storage. I think it would be most clear to fail with a `BadOrigin` error, as is the case now, rather than with a `DidNotPresent`. What do you think?